test/yagna/module/payments/test_zero_amount_txs.py

```diff
@@ -0,0 +1,142 @@
+    PROXY_HOST,
+    YAGNA_REST_URL,
+)
+from goth.node import node_environment
+from goth.runner import Runner
+from goth.runner.container.payment import PaymentIdPool
+from goth.runner.container.yagna import YagnaContainerConfig
+from goth.runner.provider import ProviderProbeWithLogSteps
+from goth.runner.requestor import RequestorProbeWithApiSteps
+
+from ya_payment import InvoiceStatus
+
+logger = logging.getLogger(__name__)
+
+
+def _topology(
+    assets_path: Path, payment_id_pool: PaymentIdPool
+) -> List[YagnaContainerConfig]:
+    # Nodes are configured to communicate via proxy
+    provider_env = node_environment(
+        rest_api_url_base=YAGNA_REST_URL.substitute(host=PROXY_HOST),
+    )
+    requestor_env = node_environment(
+        rest_api_url_base=YAGNA_REST_URL.substitute(host=PROXY_HOST),
+    )
+
+    provider_volumes = {
+        assets_path
+        / "provider"
+        / "presets.json": "/root/.local/share/ya-provider/presets.json"
+    }
+
+    return [
+        YagnaContainerConfig(
+            name="requestor",
+            probe_type=RequestorProbeWithApiSteps,
+            volumes={assets_path / "requestor": "/asset"},
+            environment=requestor_env,
+            payment_id=payment_id_pool.get_id(),
+        ),
+        YagnaContainerConfig(
+            name="provider",
+            probe_type=ProviderProbeWithLogSteps,
+            environment=provider_env,
+            # https://github.com/golemfactory/goth/issues/410
+            privileged_mode=True,
+            volumes=provider_volumes,
+        ),
+    ]
+
+
+@pytest.mark.asyncio
+async def test_zero_amount_invoice_is_settled(
+    assets_path: Path,
+    demand_constraints: str,
+    exe_script: dict,
+    payment_id_pool: PaymentIdPool,
+    runner: Runner,
+    task_package_template: str,
+):
+    """Test successful flow requesting WASM tasks with goth REST API client."""
+
+    topology = _topology(assets_path, payment_id_pool)
+
+    async with runner(topology):
+        requestor = runner.get_probes(probe_type=RequestorProbeWithApiSteps)[0]
+        provider = runner.get_probes(probe_type=ProviderProbeWithLogSteps)[0]
+
+        # Market
+
+        await provider.wait_for_offer_subscribed()
+
+        task_package = task_package_template.format(
+            web_server_addr=runner.host_address, web_server_port=runner.web_server_port
+        )
+
+        subscription_id, demand = await requestor.subscribe_template_demand(
+            task_package, demand_constraints
+        )
+
+        proposal = (
+            await requestor.wait_for_proposals(
+                subscription_id,
+                [provider],
+                lambda proposal: (
+                    proposal.properties.get("golem.runtime.name") == "wasmtime"
+                ),
+            )
+        )[0]
+        logger.info("Collected the proposal")
+
+        logger.info("Processing proposal from %s", provider.name)
+
+        # HINT: According to @kubam this is needed.
```

Comment:
This _is_ required, otherwise the test fails. No need to leave a comment here I think.